### PR TITLE
feat(localdevelopment): can now use blockchain for local dev market testing

### PR DESCRIPTION
**Pull request by bilbolPrime**
### All Submissions

* [ ] Can you post a screenshot of your changes (if applicable)?
No UI changes

### New Feature Submissions

* [ ] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?
* [ ] #363 

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description

Added a flag VUE_APP_FEATURE_FLAG_MARKET_USE_BLOCKCHAIN that, when set to true, switches to market to local chain to allow devs to work with market.
Tried to keep it as clean as possible but some ugly ifs and elses were inevitable.

**Could not test API locally, please allow a good buffer on testnet before going to prod so that no bugs arise on prod.**

### CODE REVIEW

File: frontend/src/views/Market.vue [hunk 1]
```diff
@@ -675,6 +676,21 @@ export default Vue.extend({
       this.marketOutcome = null;
       this.waitingMarketOutcome = true;
 
+      if(useBlockchain === true)
```
Comment (seiyria) [inline frontend/src/views/Market.vue:679]: `=== true` is superfluous in this case

Comment (seiyria) [reply frontend/src/views/Market.vue:679]: always use braces, no exceptions

File: frontend/src/views/Market.vue [hunk 2]
```diff
@@ -762,17 +813,26 @@ export default Vue.extend({
       this.waitingMarketOutcome = true;
 
       try {
-        const results = this.activeType === 'weapon' ?
-          await this.searchWeaponListingsBySeller(this.search):
-          await this.searchCharacterListingsBySeller(this.search);
-        this.searchResults = results;
+        if(useBlockchain === true){
```
Comment (seiyria) [inline frontend/src/views/Market.vue:816]: `=== true is superfluous`

File: frontend/src/views/Market.vue [hunk 3]
```diff
@@ -787,13 +847,20 @@ export default Vue.extend({
 
       this.waitingMarketOutcome = true;
 
-      const results = this.activeType === 'weapon' ?
-        await this.searchWeaponListingsBySeller(this.defaultAccount) :
-        await this.searchCharacterListingsBySeller(this.defaultAccount);
+      if(useBlockchain === true){
```
Comment (seiyria) [inline frontend/src/views/Market.vue:850]: `=== true`

File: frontend/src/views/Market.vue (hunk 2 above)
Comment (seiyria) [reply frontend/src/views/Market.vue:816]: split this into two functions

File: frontend/src/views/Market.vue (hunk 3 above)
Comment (seiyria) [reply frontend/src/views/Market.vue:850]: split this into two functions

**Review by seiyria — CHANGES_REQUESTED** [review]
I think instead of having blockchain checks in the relevant functions, I'd split that function into two, and call the respective function. This is done in a few places but not everywhere.

File: frontend/src/views/Market.vue (hunk 1 above)
Comment (bilbolPrime) [reply frontend/src/views/Market.vue]: done

File: frontend/src/views/Market.vue (hunk 3 above)
Comment (bilbolPrime) [reply frontend/src/views/Market.vue]: done

File: frontend/src/views/Market.vue (hunk 2 above)
Comment (bilbolPrime) [reply frontend/src/views/Market.vue]: done

**Review by seiyria — APPROVED** [review]

**Pull request merged**